src/jit/lsra.cpp

```diff
@@ -4070,6 +4075,15 @@ void LinearScan::buildRefPositionsForNode(GenTree*                  tree,
             useCandidates  = allRegs(registerType);
         }
 
+#ifdef _TARGET_ARM_
+        if (tree->OperIsPutArgStruct())
+        {
+            // get i-th candidate
+            currCandidates = genFindLowestReg(candidates);
```

Comment:
Ah, I was confused. Instead, how about asserting that the register is `genRegArgNext` of the previous register, if any?